src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event details
+    /// </summary>
+    public class AzureMaintenanceEvent
+    {
+        internal AzureMaintenanceEvent(string message)
+        {
+            RawMessage = message;
+            try
+            {
+                var info = message?.Split('|');
+                for (int i = 0; i < info?.Length / 2; i++)
+                {
+                    string key = null, value = null;
+                    if (2 * i < info.Length) { key = info[2 * i].Trim(); }
+                    if (2 * i + 1 < info.Length) { value = info[2 * i + 1].Trim(); }
+                    if (!string.IsNullOrEmpty(key) && !string.IsNullOrEmpty(value))
+                    {
+                        switch (key.ToLowerInvariant())
+                        {
+                            case "notificationtype":
+                                NotificationType = value;
+                                break;
+
+                            case "starttimeinutc":
+                                if (DateTime.TryParse(value, out DateTime startTime))
+                                {
+                                    StartTimeUtc = DateTime.SpecifyKind(startTime, DateTimeKind.Utc);
+                                }
+                                break;
+
+                            case "isreplica":
+                                bool.TryParse(value, out IsReplica);
+                                break;
+
+                            case "ipaddress":
+                                IPAddress.TryParse(value, out IpAddress);
+                                break;
+
+                            case "sslport":
+                                Int32.TryParse(value, out var port);
+                                SSLPort = port;
+                                break;
+
+                            case "nonsslport":
+                                Int32.TryParse(value, out var nonsslport);
+                                NonSSLPort = nonsslport;
+                                break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch { }
```

Comment:
I'm worried we'll silently throw here and not have any debuggability, but we don't really have an out for it today. One proposal (not for this PR) was to have a rolling log for debugging. Can we please add a comment here for later in the catch? e.g. `// TODO: Append to rolling debug log when it's present`

Reply:
Absolutely! I was unsure whether or not to add a comment here. Thanks for clarifying